cluster/src/main/java/org/apache/iotdb/cluster/client/sync/SyncClientAdaptor.java

```diff
@@ -349,6 +351,26 @@ public static Integer getPathCount(AsyncDataClient client, Node header, List<Str
     return handler.getResult(RaftServer.getReadOperationTimeoutMS());
   }
 
+  public static Set<String> getDevices(AsyncDataClient client, Node header, ShowDevicesPlan plan)
+      throws InterruptedException, TException, IOException {
+    GetDevicesHandler handler = new GetDevicesHandler();
+    AtomicReference<Set<String>> response = new AtomicReference<>(null);
+    handler.setResponse(response);
+    handler.setContact(client.getNode());
+    ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+    DataOutputStream dataOutputStream = new DataOutputStream(byteArrayOutputStream);
+    plan.serialize(dataOutputStream);
+
+    client.getDevices(header, ByteBuffer.wrap(byteArrayOutputStream.toByteArray()),
+        handler);
+    synchronized (response) {
```

Comment:
`response` is a local variants and will create a new instance when getDevices is called, so 
synchronized response seems unnecessary

Reply:
I think `GetDevicesHandler` has the answer.